examples_tests/48.ArithmeticUnitTest/main.cpp

```diff
@@ -0,0 +1,230 @@
+		infos[0].desc = gpuinputDataBuffer;
+		infos[0].buffer = { 0u, BUFFER_SIZE };
+		for (size_t i = 1; i <= 7; i++)
+		{
+			infos[i].desc = buffers[i - 1];
+			infos[i].buffer = { 0u,BUFFER_SIZE };
+
+		}
+		IGPUDescriptorSet::SWriteDescriptorSet writes[8];
+		for (uint32_t i = 0u; i < 8; i++)
+			writes[i] = { descriptorSet.get(),i,0u,1u,EDT_STORAGE_BUFFER,infos + i };
+		driver->updateDescriptorSets(8, writes, 0u, nullptr);
+	}
+	struct GLSLCodeWithWorkgroup {
+		uint32_t workgroup_definition_position;
+		std::string glsl;
+	};
+	constexpr const char* symbolsToReplace = "????";
+	auto getShaderGLSL = [&](const char* filePath)
+	{
+		std::ifstream file(filePath);
+		std::stringstream buff; buff << file.rdbuf();
+		std::string shaderCode = buff.str();
+		uint32_t wgPos = shaderCode.find(symbolsToReplace, 0);
+		GLSLCodeWithWorkgroup ret = { wgPos,shaderCode };
+		return ret;
+	};
+	GLSLCodeWithWorkgroup shaderGLSL[3] =
+	{
+		getShaderGLSL("../testReduce.comp"),
+		getShaderGLSL("../testInclusive.comp"),
+		getShaderGLSL("../testExclusive.comp"),
+	};
+
+
+auto getGPUShader = [&](GLSLCodeWithWorkgroup glsl, uint32_t wg_count)
+{
+	auto alteredGLSL = glsl.glsl.replace(glsl.workgroup_definition_position, 4, std::to_string(wg_count));
+	auto shaderUnspecialized = core::make_smart_refctd_ptr<asset::ICPUShader>(alteredGLSL.data());
+	asset::ISpecializedShader::SInfo specinfo(nullptr, nullptr, "main", IGPUSpecializedShader::ESS_COMPUTE,"../file.comp");
+	auto cs = core::make_smart_refctd_ptr<asset::ICPUSpecializedShader>(std::move(shaderUnspecialized), std::move(specinfo));
+	auto cs_rawptr = cs.get();
+	core::smart_refctd_ptr<IGPUSpecializedShader> shader = driver->getGPUObjectsFromAssets(&cs_rawptr, &cs_rawptr + 1)->front();
+	return shader;
+};
+	for (size_t current_Workgroup = 2; current_Workgroup < 1024; current_Workgroup++)
+	{
+
+
+		auto reducePipeline = driver->createGPUComputePipeline(nullptr, core::smart_refctd_ptr(pipelineLayout), std::move(getGPUShader(shaderGLSL[0], current_Workgroup)));
+		auto inclusivePipeline = driver->createGPUComputePipeline(nullptr, core::smart_refctd_ptr(pipelineLayout), std::move(getGPUShader(shaderGLSL[0], current_Workgroup)));
+		auto exclusicePipeline = driver->createGPUComputePipeline(nullptr, core::smart_refctd_ptr(pipelineLayout), std::move(getGPUShader(shaderGLSL[0], current_Workgroup)));
+
+
+		core::smart_refctd_ptr<IGPUComputePipeline> pipelines[3] = { reducePipeline ,inclusivePipeline ,exclusicePipeline };
+
+
+		if (device->run())
+		{
+			driver->beginScene(true);
+			for (size_t i = 0; i < 3; i++)	//1 because skipping reduce! UNDO THIS
+			{
+				driver->bindComputePipeline(pipelines[i].get());
+				const video::IGPUDescriptorSet* ds = descriptorSet.get();
+				driver->bindDescriptorSets(video::EPBP_COMPUTE, pipelines[i]->getLayout(), 0u, 1u, &ds, nullptr);
+				driver->dispatch(BUFFER_SIZE / (sizeof(uint32_t)), 1, 1);
+				video::COpenGLExtensionHandler::extGlMemoryBarrier(GL_ALL_BARRIER_BITS);
```

Comment:
note for the future, because you're only using the buffer for a copy and then reusing it as an ssbo, only the following combination of flags is requires
`GL_BUFFER_UPDATE_BARRIER_BIT|GL_SHADER_STORAGE_BARRIER_BIT`